pkg/init/usermode-helper.c

```diff
@@ -0,0 +1,36 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[])
+{
+	if (!strcmp(argv[0], "/sbin/mdev")) {
+		/* busybox uses /sbin/mdev for early uevent bootstrapping */
+		execv(argv[0], argv);
+	} else if (!strcmp(argv[0], "/sbin/modprobe")) {
+		/* allow modprobe */
+		execv(argv[0], argv);
+	} else {
+		/* This means either we got an unexpected call from the kernel
+		 * or someone is doing something nefarious. Some other possible
+		 * expected callers are:
+		 *  - for core dumps. we don't have a "core" binary, and don't
+		 *    set this by default to anything. when we do, we need to
+		 *    whitelist it here
+		 *  - /linuxrc: we're not doing legacy root setup, so we don't
+		 *     need this
+		 *  - a few drivers and filesystems (drbd, nfs, nfsd, ocfs2)
+		 *  - cgroup notify_on_release handlers, which we do not set
+		 *    (but e.g. systemd needs, if anyone ever tries to boot
+		 *    that on linuxkit)
+		 *  - /sbin/request-key, which we don't provide
+		 *  - on x86, machine check
+		 *
+		 * Today we only call mdev and modprobe, but as we add more
+		 * features to linuxkit this whitelist may need changing (or a
+		 * policy, like always allow stuff in /sbin).
+		 */
+		exit(2);
```

Comment:
[projects/logging](https://github.com/linuxkit/linuxkit/tree/master/projects/logging) contains an in-memory logging daemon that could capture the logs. I've not made it syslog compatible (yet) as it captures data from fd's passed to it (typically stderr/stdout from onboot/service/dockerd containers) so we don't have to bind mount /dev/log, but if we need it we could add it. It's a bit hard to enable at the moment unfortunately as it needs to run as a service before the onboot containers, which requires a modified `init`